inclusive-language.md

```diff
@@ -0,0 +1,56 @@
+- Feature Name: Inclusive Language
+- Start Date: 2020-10-05
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Following examples set by others in the tech industry and even the Linux kernel itself, Salt should adopt inclusive language to replace racially-charged jargon.
+
+# Motivation
+[motivation]: #motivation
+
+**Why are we doing this?** If Salt is to continue to grow and gain wider adoption, it needs to drop references to terms that do not have a place in this world.
+
+**What use cases does it support?** This will affect all current and future users of Saltstack.
+
+**What is the expected outcome?** Salt will join a growing number of companies and projects that are making changes to use inclusive language.
+
+**If this SEP is not accepted?** I don't believe there is an alternative way around this.
+
+# Design
+[design]: #detailed-design
+
+**Proposed Updated Terminology**
+Salt does use non-inclusive language for a key component. While I will submit what is becoming commonly accepted alternatives, I think the Salt Master/Minon terms will require some larger considerations. I understand that it's not just updating a configuration term, but falls into the larger product branding. I understand this will likely touch a lot of files, but should not require any significant refactoring of code. Mostly just agreeing upon new terms and making sure they mach up as they are updated. 
+
+| Avoid non-inclusive language | Prefer inclusive versions |
+| ---------------------------- | ------------------------- |
+| Whitelist | Allowlist |
+| Blacklist | Blocklist |
+| Master/slave | Leader/follower, primary/replica, primary/standby |
```

Comment:
Salt does not use the word "slave". None of the preferred terms accurately describe the role of the master.

Reply:
I tried to include a list of terms that have been provided by other companies as general alternatives. I understand that in this context, the alternative names do not provide an accurate label. While "slave" is not included in any of Salt's language, the term "master" ,on its own, is enough. Having to have a conversation about why a word isn't racist, right or wrong in the given context, highlights the need to change it. A possible alternative name might be "commander". This could be tied into the branding as well, "Salt Commander, helping lead the good fight in SecOps".

Reply:
> right or wrong in the given context

So we should just accept the wrong thing as being true? Or pander to minority groups by making mindless, meaningless changes?

Reply:
> > right or wrong in the given context
> 
> So we should just accept the wrong thing as being true? Or pander to minority groups by making mindless, meaningless changes?

"mindless, meaningless changes" feels like a strong statement with many other software companies seeing value in removing these terms. Even Github themselves now prefer main vs master https://github.com/github/renaming/

VMware seems to have an opinion, which will hopefully impact Salt https://www.vmware.com/company/news/updates/replacing-problematic-terminology.html

Reply:
> VMware seems to have an opinion

Hopefully that means we don't have to argue about it. There's no need for an SEP if they just make the decision.

Reply:
Whether **you** think it's right or wrong is not the way to look at this.  The use of the word master in this case (a server controlling other devices) has a negative association and is harmful to others.  Also, using black/white to describe bad/good, is just as bad.  This is not about how you see the use of the word, it's how others feel about the use of the word.

https://en.wikipedia.org/wiki/Master/slave_(technology)
https://tools.ietf.org/id/draft-knodel-terminology-01.html
https://www.clockwork.com/news/creating-inclusive-naming-conventions-in-technology/
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb
https://developers.google.com/style/inclusive-documentation
https://medium.com/fact-of-the-day-1/inclusive-naming-conventions-77e468430216
https://www.cio.com/article/2438337/luser--pebkac-and-other-ways-it-insults-users.html
https://www.politico.com/news/2020/06/25/agency-ends-use-technology-terms-racist-associations-339880

Reply:
Where does this end though? Couldn't any word be considered offensive to someone?

Reply:
While I agree with the sentiment around master in the context of Salt doesn't evoke comparisons to a very controversial past, there's no doubt a not-insignificant number of people do (regardless of context). Might I suggest the most appropriate alternative be "director?"

Reply:
To continue the bikeshedding a bit, and maybe explore some history, I'll throw in my $0.02USD.

Salt has *never* used "master/slave" terminology, AFAIK, but has been "master/minion". The [origins of minion](https://www.grammarphobia.com/blog/2020/08/henchmen-minions.html) seem to come from the French "mignon" -- which meant "a king's favorite". Eventually it came to just mean a hanger-on, a follower.

Master, in the Salt master/minion relationship definitely applies to the 14c use of ["one who has the power to control, use, or dispose at will"](https://www.etymonline.com/word/master). Obviously the word also has strong ties to slavery, especially American slavery. As @ggiesen rightly states, for many people the word "master" evokes those ties -- so it is worth avoiding.

If the term master is off the table then, what's the most accurate term we can use to replace it? Well, the [Moby thesaurus has a multitude of options](http://dict.org/bin/Dict?Form=Dict2&Database=moby-thesaurus&Query=master), but I'll just provide a handful:

- boss
- captain
- chief
- commander
- controller
- director (suggested by @ggiesen)
- executive
- instructor
- maven
- owner
- proprietor
- wizard

The fantasy fan in me wouldn't mind a wizard casting spells, but a change like that definitely requires an alternate reality :trollface: Out of the rest of these, _controller_ seems most accurate to me. Director is a good choice, but at least in my experience a director's instructions can be flexible or ambiguous. In Salt, that's not the case. Instructions sent to the minions **must** be followed precisely and accurately. The computers must be controlled. Which sounds a bit like the beginning of a dystopian sci-fi movie, TBH :joy: 

Anyway - those are my thoughts on the color of this bikeshed :upside_down_face:

Reply:
Internally we called out setup salt-director. moving away from salt-master needs to happen IMHO

Reply:
Yeah I thought "director" might be a good term because of synergies with other VMware products (vCloud Director). Of course, the opposite might be true and there could be concern about confusion

Reply:
I really like `salt-hero` as a potential, too.

Working with **heroes** makes it sound like we all have superpowers when using `salt`.